### PR TITLE
📝 Add documentation on the RumUserActionDetector

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What and why?

RumUserActionDetector has been available as part of the API for a while and was recently improved to have more functionality. Adding documentation on how to use it to the main docs site.

### Review checklist

- [ ] This pull request has appropriate unit and / or integration tests 
- [ ] This pull request references a Github or JIRA issue

### CI Configuration (optional)
- [ ] Run unit tests
- [ ] Run integration tests

### CODE REVIEW

File: packages/datadog_flutter_plugin/doc/rum/advanced_configuration.md [hunk 1]
```diff
@@ -10,23 +10,56 @@ Flutter RUM automatically tracks attributes such as user activity, views (using
...
+        rum: DatadogSdk.instance.rum,
+        child: Scaffold(
+          // etc.
+        ),
+      ),;
+    )
+  }
+```
+
+`RumUserActionDetector` will detect tap actions with certan standard Flutter widgets including `ElevatedButton`, `TextButton`, `CupertinoButton`, `BottomNavigationBar`, `TabBar`, `InkWell`, and `GestureDetector`.
```
Comment (hestonhoffman) [inline packages/datadog_flutter_plugin/doc/rum/advanced_configuration.md:46]:
```suggestion
`RumUserActionDetector` detects tap actions with certain standard Flutter widgets including `ElevatedButton`, `TextButton`, `CupertinoButton`, `BottomNavigationBar`, `TabBar`, `InkWell`, and `GestureDetector`.
```

File: packages/datadog_flutter_plugin/doc/rum/advanced_configuration.md [hunk 2]
```diff
@@ -10,23 +10,56 @@ Flutter RUM automatically tracks attributes such as user activity, views (using
...
+
+For most button types, the detector will look for a `Text` widget child to use as the description of the action. In order cases, it will look for a child `Semantics` object, or an `Icon` whith its `Icon.semanticlsLabel` proprety set. If you want more fine grained control of the description of the action, you can also enclose your Widget tree in a [`RumUserActionAnnotation`][6]. All interactions with supported Flutter widgets under the Annotation will use its description.
```
Comment (hestonhoffman) [inline packages/datadog_flutter_plugin/doc/rum/advanced_configuration.md:48]:
```suggestion
For most button types, the detector looks for a `Text` widget child to use as the description of the action. In other cases, it looks for a child `Semantics` object, or an `Icon` with its `Icon.semanticlsLabel` property set. If you want more fine grained control of the description of the action, you can enclose your Widget tree in a [`RumUserActionAnnotation`][6]. All interactions with supported Flutter widgets under the Annotation use its description.
```

File: packages/datadog_flutter_plugin/doc/rum/advanced_configuration.md [hunk 3]
```diff
@@ -10,23 +10,56 @@ Flutter RUM automatically tracks attributes such as user activity, views (using
...
+  RumUserActionAnnotation(
+    description: 'Add Checkout Item',
+    child: FloatingActionButton(
+      child: const Icon(Icons.add)
+      // etc.
+    ),
+  )
+```
+
+To track more complex interactions, or to track specific user actions manually such as scrolls or swipe, you can use `DdRum.addUserAction`, `DdRum.startUserAction` and `DdRum.stopUserAction`.
```
Comment (hestonhoffman) [inline packages/datadog_flutter_plugin/doc/rum/advanced_configuration.md:62]:
```suggestion
To track more complex interactions, or to track specific user actions manually such as scrolls or swipes, you can use `DdRum.addUserAction`, `DdRum.startUserAction` and `DdRum.stopUserAction`.
```

File: packages/datadog_flutter_plugin/doc/rum/advanced_configuration.md [hunk 4]
```diff
@@ -224,5 +256,7 @@ This means that even if users open your application while offline, no data is lo
 [2]: https://docs.datadoghq.com/real_user_monitoring/flutter/#setup
 [3]: https://docs.datadoghq.com/real_user_monitoring/flutter/data_collected
 [4]: https://docs.datadoghq.com/real_user_monitoring/explorer/?tab=measures#setup-facets-and-measures
-[5]: https://github.com/DataDog/dd-sdk-flutter/tree/main/packages/datadog_tracking_http_client
-[6]: https://pub.dev/documentation/datadog_flutter_plugin/latest/datadog_flutter_plugin/
\ No newline at end of file
+[5]: https://pub.dev/documentation/datadog_flutter_plugin/latest/datadog_flutter_plugin/RumUserActionDetector-class.html
+[6]: https://pub.dev/documentation/datadog_flutter_plugin/latest/datadog_flutter_plugin/RumUserActionAnnotation-class.html
+[7]: https://github.com/DataDog/dd-sdk-flutter/tree/main/packages/datadog_tracking_http_client
+[8]: https://pub.dev/documentation/datadog_flutter_plugin/latest/datadog_flutter_plugin/
```
Comment (hestonhoffman) [inline packages/datadog_flutter_plugin/doc/rum/advanced_configuration.md:262]:
```suggestion
[8]: https://pub.dev/documentation/datadog_flutter_plugin/latest/
```

**Review by hestonhoffman — COMMENTED** [review]
Left some edits

**Review by 0xnm — APPROVED** [review]

**Review by hestonhoffman — APPROVED** [review]

**Pull request merged**